pkg/minikube/machine/start.go

```diff
@@ -265,10 +265,14 @@ func postStartSetup(h *host.Host, mc config.ClusterConfig) error {
 }
 
 // acquireMachinesLock protects against code that is not parallel-safe (libmachine, cert setup)
-func acquireMachinesLock(name string) (mutex.Releaser, error) {
+func acquireMachinesLock(name string, drver string) (mutex.Releaser, error) {
 	spec := lock.PathMutexSpec(filepath.Join(localpath.MiniPath(), "machines"))
 	// NOTE: Provisioning generally completes within 60 seconds
 	spec.Timeout = 15 * time.Minute
+	if driver.IsKIC(drver) {
+		spec.Timeout = 20 * time.Second
```

Comment:
I expect that this will cause a great deal of integration testing instability.

My suggestion would be to change this lock for KIC so that the `spec` includes the machine name at the end. As best as I can tell, Docker does not have any problems with creating multiple containers simultaneously.

With that change, 120 seconds is probably fine for Docker. Without it, it will mean unnecessary crashing in integration tests, as `acquireMachinesLock` shares locks across machines.